docs/src/pages/components/data-grid/demo/FullFeauredDemo.tsx

```diff
@@ -0,0 +1,148 @@
+import * as React from 'react';
+import { XGrid, GridToolbar } from '@material-ui/x-grid';
+import { useDemoData } from '@material-ui/x-grid-data-generator';
+import KeyboardArrowRightIcon from '@material-ui/icons/KeyboardArrowRight';
+import { makeStyles } from '@material-ui/core/styles';
+import FormControl from '@material-ui/core/FormControl';
+import FormGroup from '@material-ui/core/FormGroup';
+import Button from '@material-ui/core/Button';
+import FormLabel from '@material-ui/core/FormLabel';
+import MenuItem from '@material-ui/core/MenuItem';
+import Select from '@material-ui/core/Select';
+
+const useStyles = makeStyles((theme) => ({
+  root: {
+    display: 'flex',
+    flexDirection: 'column',
+    height: 600,
+    width: '100%',
+    '& .MuiFormGroup-options': {
+      alignItems: 'center',
+      justifyContent: 'center',
+      marginBottom: theme.spacing(2),
+      '& fieldset': {
+        padding: theme.spacing(0, 1),
+      },
+      '& button': {
+        marginLeft: theme.spacing(3),
+      },
+    },
+  },
+}));
+
+function SettingsPanel(props) {
+  // eslint-disable-next-line react/prop-types
```

Comment:
We could potential use TypeScript here, but no real perferences

Reply:
That one actually is not needed. I do need to generate the propTypes tho because the build is failing because of this.

Reply:
> I do need to generate the propTypes 

Well, it's not directly you that generate them, but the script that generate the JS demos from the TS one :)

Reply:
Yes, If the code is ok I'll go ahead and fix/type the `tsx` file.

Reply:
Things I have noticed:

- Ant Design is not spelled correctly
- On mobile, the form field are too close to each other
- It could be interesting to enable the pagination, with pages of 1,000 rows, not sure, only an idea.
- In the future we can improve the useDemoData hook. Have it create the records during the idle time, not blocking the JavaScript thread, as this makes the task asynchronous, add a loading state (loading could be a skeleton, it doesn't have to be circular progress).

It would be great to have @dtassone review on this.

Reply:
the issue with yarn lint is due to the regressions tests, just need to exclude the folders... I put a fix in my autopagesize pr

Reply:
> the issue with yarn lint is due to the regressions tests, just need to exclude the folders... I put a fix in my autopagesize pr

If have left a review on the changes. IMHO they could have been in a standalone PR.

I personally never run eslint locally, sorry, I didn't catch it.

Reply:
I created a PR

Reply:
rebase and merge to fix the eslint issue